### ISSUE TITLE
Create temporary end-to-end suite of test cases

### ISSUE THREAD

**Issue by spencerkclark** [spencerahill/aospy#42]
In lieu of a robust set of unit tests (see #22), which will take time to develop, we should have a set of aospy objects (and associated computations) that span the current horizon of the package's uses.  Ideally we'd be able to run a script that would attempt to preform all the specified computations on the set of objects;  if it ran successfully we'd at least have some confidence that it was doing what we wanted (or we'd be able to see what tripped it up).  

The details are still a bit up in the air (we'd have to figure out how to get things to run with one click), but for now I suggest we create a Project that contains Runs from some of the Models we have worked with:
- [x] AM2
- [x] AM3
- [x] HiRAM
- [x] Idealized Moist
- [x] CMIP5 output

We'd then have to come up with a set of computations to do on each of them.  ~~To start out I think these can just be computations that aospy can do without a call to an external `aospy_user/calcs/` function (although I guess there is no reason we couldn't have some dummy functions within this pseudo-aospy_user environment).~~ (This was addressed in #61, which now does allow for calls to external calculation functions located within `test_objs/calcs.py`).  The most basic computations would be (feel free to add more as they come up):
- Annual mean, and time series
- Seasonal mean, and time series
- Monthly mean, and time series
- Vertical integral over interpolated pressure levels
- Vertical integral over sigma pressure surfaces
- Regional average, and time series (with no land mask)
- Regional average, and time series (with land mask)
- Regional average, and time series (with ocean mask)

It would also be great if we could have some reference computations to compare to for these calculations, but that too might be a goal for down the road.  Initially it can just answer the question "does it run or not?"

**Comment by spencerahill** [comment]
How do you feel about this in light of #61?

**Comment by spencerkclark** [comment]
With regard to potentially closing this issue?  Should we wait until we have more coverage, or do you think #61 is good enough for now?  

On the bright side, with #61 it shouldn't be too hard to check a lot of these things off the list.  I can do my part in adding some more tests on AM2 and idealized model output, but I don't have any AM3, HiRAM, or CMIP5 runs handy.  In principle, though, I could just lift some from your object library.

**Comment by spencerahill** [comment]
Ya, just wanted to see if this still has merits, which it does.  I agree, it would be useful (and low-hanging fruit) to copy the other model objects here, since often in the past a change would break things for one model and not the other.

(Doing these does _not_ obviate the need for more unit testing everywhere, of course.)

**Comment by spencerkclark** [comment]
:+1: I'll try and get to this soon.

**Comment by spencerahill** [comment]
Awesome (no rush though if you have generals/other commitments)

**Comment by spencerahill** [comment]
I think the best way of extending #64 to other models will be to abstractify all of these test methods you've created for AM2 by moving them into a separate class that accepts a generic model object, and then creating classes for each model that inherits from that class and provides the needed model object; the tests will then automatically run for each model.  See infinite-diff's tests for something of an example: the [TestBwdDiff](https://github.com/spencerahill/infinite-diff/blob/develop/infinite_diff/test/test_finite_diff.py#L100) class inherits from TestFwdDiff and just supplies a different `self.method`, which the tests then use.  (And the latter inherits from a generic class that supplies misc. arrays and other attributes to use as the test data.)

(FYI I'm in the midst of refactoring that package, so that link might not work after the next few days, but the concept is relatively straightforward.)

**Comment by spencerkclark** [comment]
That's a really good idea.  I'll do that in my next pull request.

**Comment by spencerahill** [comment]
:+1: I am starting to get a better feel for OOP and polymorphism...it is really powerful.

**Comment by spencerahill** [comment]
So am I correct that after #66 we are all set on this except for CMIP5 (which I need to figure out)?

**Comment by spencerkclark** [comment]
Yes, that's correct, and for what it's worth the CMIP5 tests are implemented (albeit they fail), but perhaps that's a separate issue (cf. spencerahill/aospy-obj-lib#10).

**Comment by spencerahill** [comment]
Right.  In that case, let's call this done!  I'll re-open if, once I get the CMIP5 ones sorted out, these tests don't work.

Thanks for all your work on this!
